dataikuapi/dss/ml.py

```diff
@@ -308,50 +332,679 @@ def set_metric(self, metric=None, custom_metric=None, custom_metric_greater_is_b
+        assert isinstance(range_bound, int) or isinstance(range_bound, float), \
+            "Invalid input type for hyperparameter \"{}\": ".format(hyperparameter_name) + \
+            "range bounds must be numbers"
+
+
+class RandomForestSettings(AlgorithmSettings):
+
+    def __init__(self, raw_settings, hyperparameter_search_params):
+        super(RandomForestSettings, self).__init__(raw_settings, hyperparameter_search_params)
+        self.n_estimators = NumericalHyperparameterSettings("n_estimators", self)
+        self.min_samples_leaf = NumericalHyperparameterSettings("n_estimators", self)
+        self.max_tree_depth = NumericalHyperparameterSettings("max_tree_depth", self)
+        self.max_feature_prop = NumericalHyperparameterSettings("max_feature_prop", self)
+        self.max_features = NumericalHyperparameterSettings("max_features", self)
+        self.n_jobs = SingleValuedHyperparameterSettings("n_jobs", self)
+        self.selection_mode = SingleValuedHyperparameterSettings("selection_mode", self, ["auto", "sqrt", "log2", "number","prop"])
+
+
+class XGBoostSettings(AlgorithmSettings):
+
+    def __init__(self, raw_settings, hyperparameter_search_params):
+        super(XGBoostSettings, self).__init__(raw_settings, hyperparameter_search_params)
+        self.max_depth = NumericalHyperparameterSettings("max_depth", self)
+        self.learning_rate = NumericalHyperparameterSettings("learning_rate", self)
+        self.gamma = NumericalHyperparameterSettings("gamma", self)
+        self.min_child_weight = NumericalHyperparameterSettings("min_child_weight", self)
+        self.max_delta_step = NumericalHyperparameterSettings("max_delta_step", self)
+        self.subsample = NumericalHyperparameterSettings("subsample", self)
+        self.colsample_bytree = NumericalHyperparameterSettings("colsample_bytree", self)
+        self.colsample_bylevel = NumericalHyperparameterSettings("colsample_bylevel", self)
+        self.alpha = NumericalHyperparameterSettings("alpha", self)
+        self.lambda_ = NumericalHyperparameterSettings("lambda", self)
+        self.booster = CategoricalHyperparameterSettings("booster", self)
+        self.objective = CategoricalHyperparameterSettings("objective", self)
+        self.n_estimators = SingleValuedHyperparameterSettings("n_estimators", self)
+        self.nthread = SingleValuedHyperparameterSettings("nthread", self)
+        self.scale_pos_weight = SingleValuedHyperparameterSettings("scale_pos_weight", self)
+        self.base_score = SingleValuedHyperparameterSettings("base_score", self)
+        self.impute_missing = SingleValuedHyperparameterSettings("impute_missing", self, [True, False])
+        self.missing = SingleValuedHyperparameterSettings("missing", self)
+        self.cpu_tree_method = SingleValuedHyperparameterSettings("cpu_tree_method", self)
+        self.gpu_tree_method = SingleValuedHyperparameterSettings("gpu_tree_method", self)
```

Comment:
For these, we have no accepted values, because it is a space delimited string, right ?

Reply:
No I just forgot about them !